weatherman.py

```diff
@@ -0,0 +1,109 @@
+"""
+Weatherman
+This module generates yearly reports on weather data collected from  Dec 1996 - May 2011.
+Report no. 1 gives the yearly max/min temperature and max/min humidity.
+Report no. 2 gives the hottest day of the year and the temperature on that day.
+"""
+
+import os
+import sys
+
+import argparse
+
+
+max_temp_ = {}
+max_temp_date = {}
+min_temp_ = {}
+max_humidity= {}
+min_humidity= {}
+
+parser = argparse.ArgumentParser(prog = 'weatherman',
+								 usage = '%(prog)s [report#] [data_dir]',
+								 description = 'Generate weather reports')
+parser.add_argument('-d', '--data_dir', type = str, choices = ['./weatherdata/'],
+					help = 'Directory containing weather data files', required = True)
+parser.add_argument('-r', '--report_number', type = int, choices = [1, 2],
+					help = '1 for Annual Max/Min Temperature, 2 for Hottest day of each year',
+					required = True)
+try:
+	args = parser.parse_args()
+except:
+    parser.print_help()
+    sys.exit(0)
+
+"""
+Used for printing yearly reports.
+User gets 2 options.
+1 for yearly max/min temperature and max/min humidity.
+2 for hottest day of the year and the temperature on that day.
+"""
+
+def print_report(report_number):
+
+	if report_number == 1:
+		print('{:18}{:15}{:16}{:17}{}'.format('Year', 'MAX Temp', 'MIN Temp', 'MAX Humidity', 'MIN Humidity'))
+		for key in sorted(max_temp_):
+			print(f'{key} {max_temp_[key]:15} {min_temp_[key]:15} {max_humidity[key]:15} {min_humidity[key]:15}')
+
+	elif report_number == 	2:
+		print('{:18}{:15}{}'.format('Year', 'MAX Temp', 'DATE'))
+		for key in sorted(max_temp_):
+			print(f'{key} {max_temp_[key]:15} {max_temp_date[key]:15}')
+
+"""
+Functions asks for the direcory path to be accessed.
+It only accepts one path which is: ./weatherdata/
+as this is the one we need to return the desired reports.
+"""
+
+def weather_directory(data_dir):
+	file_list = os.listdir(data_dir)
+	for file_name in file_list:
+
+		with open(data_dir + file_name) as data_file:
+			monthly_data = data_file.read().splitlines()[2:-1]
+			for daily_data in monthly_data:
+				daily_data = daily_data.split(',')
+				year = int(daily_data[0].split('-')[0])
+				date = daily_data[0]
+				try:
+					max_temp = int(daily_data[1])
```

Comment:
bug exists. The one explained in previous revision.

Reply:
Will work on it.

Reply:
I have corrected it, but have a little question here for confirmation.